src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/JsonConverter.Sync.cs

```diff
@@ -3,88 +3,181 @@
         }
 
-        public static T FromJson<T>(this ReadOnlySequence<byte> json, JsonConverterSettings options = null)
+        public static T FromJson<T>(this ReadOnlySequence<byte> json, JsonConverterSettings settings = null)
         {
-            return (T)FromJson(json, typeof(T), options);
+            return (T)FromJson(json, typeof(T), settings);
         }
 
-        public static object FromJson(this ReadOnlySequence<byte> json, Type returnType, JsonConverterSettings options = null)
+        public static object FromJson(this ReadOnlySequence<byte> json, Type returnType, JsonConverterSettings settings = null)
         {
             if (returnType == null)
                 throw new ArgumentNullException(nameof(returnType));
 
-            if (options == null)
-                options = s_DefaultSettings;
+            if (settings == null)
+                settings = s_DefaultSettings;
 
-            JsonReaderState state = new JsonReaderState(options.DefaultBufferSize, options.ReaderOptions);
-            Utf8JsonReader reader = new Utf8JsonReader(json, true, state);
-            return FromJsonInternal(reader, returnType, options);
+            var state = new JsonReaderState(settings.DefaultBufferSize, settings.ReaderOptions);
+            var reader = new Utf8JsonReader(json, true, state);
+            return FromJsonInternal(reader, returnType, settings);
         }
 
-        private static object FromJsonInternal(this Utf8JsonReader reader, Type returnType, JsonConverterSettings options = null)
+        private static object FromJsonInternal(this Utf8JsonReader reader, Type returnType, JsonConverterSettings settings = null)
         {
-            if (options == null)
-                options = s_DefaultSettings;
+            if (settings == null)
+                settings = s_DefaultSettings;
 
-            JsonObjectState current = default;
-            List<JsonObjectState> previous = null;
+            List<FromJsonObjectState> previous = null;
             int arrayIndex = 0;
 
-            ReadData(ref reader, options, returnType, ref current, ref previous, ref arrayIndex);
+            FromJsonObjectState current = default;
+            JsonClassInfo classInfo = settings.GetOrAddClass(returnType);
+            current.ClassInfo = classInfo;
+            if (classInfo.ClassType != ClassType.Object)
+            {
+                current.PropertyInfo = classInfo.GetPolicyProperty();
+            }
+
+            FromJson(ref reader, settings, returnType, ref current, ref previous, ref arrayIndex);
 
             return current.ReturnValue;
         }
+
+        public static string ToJsonString(object value, JsonConverterSettings settings = null)
+        {
+            if (value == null)
+                throw new ArgumentNullException(nameof(value));
+
+            Span<byte> jsonBytes = ToJsonInternal(value, settings);
+#if BUILDING_INBOX_LIBRARY
+            string stringJson = s_utf8Encoding.GetString(jsonBytes);
```

Comment:
This ifdef'd code is used quite often. Move to a static helper method to avoid duplication?

Reply:
It is used twice here. I re-added the helper method but renamed the helper class to JsonReaderHelper to match corefx.